File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/dependencies/database/DatabaseHelper.java
package com.tinlone.startupexamplejava.dependencies.database;

import com.tinlone.startupexamplejava.dependencies.Logger;

public class DatabaseHelper implements IDatabaseHelper {

    private static String config = null;

    private DatabaseHelper() {
        Logger.log("loglog", "DatabaseHelper initialized");
    }

    public static void init() {
        config = "init";
    }

    public static DatabaseHelper getInstance() {
        if (config == null){
            throw new IllegalStateException("config has no init");
        }
        return new DatabaseHelper();
    }

    @Override
    public boolean insert(Object obj) {
        return true;
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/dependencies/database/DatabaseHelper.kt
package com.tinlone.startupexamplekotlin.dependencies.database

import com.tinlone.startupexamplekotlin.dependencies.Logger
import java.lang.IllegalStateException

class DatabaseHelper private constructor() : IDatabaseHelper {
    override fun insert(obj: Any?): Boolean {
        return true
    }

    companion object {
        var config: String  ?= null

        fun init() {
            config = "init"
        }

        fun getInstance(): DatabaseHelper  {
            if (config == null){
                throw IllegalStateException("config has no init")
            }
            return DatabaseHelper()
        }
    }

    init {
        Logger.log("loglog", "DatabaseHelper initialized")
    }
}